include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -137,10 +138,13 @@ namespace Functions
     FEValues<dim> fe_v(mapping, cell->get_fe(), quad,
                        update_gradients);
     fe_v.reinit(cell);
-    std::vector< std::vector<Tensor<1,dim> > > vgrads
-    (1,  std::vector<Tensor<1,dim> >(n_components) );
+    std::vector< std::vector<Tensor<1,dim,typename VECTOR::value_type> > > vgrads
+    (1,  std::vector<Tensor<1,dim,typename VECTOR::value_type> >(n_components) );
     fe_v.get_function_gradients(data_vector, vgrads);
-    gradients = vgrads[0];
+    const unsigned int s = vgrads[0].size();
+    gradients.resize(s);
+    for (unsigned int i = 0; i < s; i++)
+      gradients[i] = vgrads[0][i];
```

Comment:
Do you have to copy individual elements because there's no copy assignment for tensors of different type?